pyetrade/market.py

```diff
@@ -1,470 +1,438 @@
-        LOGGER.debug(req.text)
-
-        if resp_format == 'json':
-            return req.json()
-        else:
-            return(req.text)  
-
-    def get_option_expire_date(self, underlier, resp_format=None):
-        ''' get_option_expiry_dates(underlier, resp_format, **kwargs)
-        
-            Return a sorted list of datetime.date objects for the underlier. Some of the returned dates may not actually
-            have any options.
-            
-            if resp_format is 'json', return the python object <== this currently doesn't work as described by the eTrade API
-            
-            Another problem is the etrade API only returns monthlies, not weeklies. Therefore, invent some weeklies which occur on Fridays
-            for the present month and one month out, generally. These may not exist, but there is no way to know without asking
-            for option_chain data and getting a NULL response.
-            
-            param: underlier
-            type: str
-            description: market symbol
-           
-            param: resp_format
-            type: str
-            description: Response format .JSON or None = XML
-           
-            Sample Request
-            GET https://etws.etrade.com/market/rest/optionexpiredate?underlier=GOOGL
-            or  https://etws.etrade.com/market/rest/optionexpiredate?underlier=GOOGL.json  <== doesn't seem to work
-            
-        '''
-
-        assert resp_format in (None,'json','xml')
-        args_str = 'underlier=%s' % underlier
-        uri = (r'market/sandbox/rest/optionexpiredate' if self.dev_environment else r'market/rest/optionexpiredate')
-        api_url = '%s/%s?%s' % (self.base_url, uri, args_str)
-        LOGGER.debug(api_url)
-        
-        if resp_format == 'json':
-            req = self.session.get(api_url + '.json')
-        else:
-            req = self.session.get(api_url)
-        req.raise_for_status()
-        LOGGER.debug(req.text)
-
-# JSON format a lot easier to deal with, but doesn't seem to work
-        if resp_format == 'json':
-            return req.json()
-        else:
-            try:
-                xmlobj = jxmlease.parse(req.text)
-                z = xmlobj['OptionExpireDateGetResponse']['ExpirationDate']
-                dates = [ dt.date(int(this_date['year']), int(this_date['month']), int(this_date['day'])) for this_date in z ]
-            except Exception as err:
-                LOGGER.error(err)
-                raise
-
-# add weeklies for this month and the following month
-        friday = NEXT_FRIDAY
-        for i in range(8):
-            if friday not in dates: dates.append(friday)
-            friday += dt.timedelta(days=7)
-        
-        return sorted(dates)
-    
+#!/usr/bin/python3
+
+"""Market - ETrade Market API V1
+
+    Generally, three different types of response that one can get from these routines are based on the requested resp_format
+        * None - return a python object

```

Comment:
@1rocketdude Is there any reason we are not using jxmlease.parse(req.text) to dump the xml and have introduced None?